### DISCUSSION TITLE
Memory leak or garbage collection?

### DISCUSSION THREAD

**Discussion by deribaucourt** [netdata/netdata#18948] [Q&A]
Hello,

Our system is going to run netdata without interruption and the memory consumption of the netdata process looks like a memory leak.

![qsd](https://github.com/user-attachments/assets/cfbb7027-0c9a-4961-92e6-7140b61401d6)

However, by looking at the live demo's profile, we can see a similar growth pattern which looks logarithmic, but then it cycles down when it reaches about 100MiB. https://app.netdata.cloud/spaces/netdata-demo/rooms/apache/overview

![image](https://github.com/user-attachments/assets/2f18902b-210d-4c2d-9dad-75650b6aeba8)

We must avoid a memory leak on our system so we would like to understand this mechanism and make sure the garbage is going to be collected regularly. Can you indicate what part of the code would produce this memory behavior and how could we change the freeing trigger to run it more regularly on our system?

Thank you for your time :)

**Comment by ilyam8** [comment]
Hi @deribaucourt, It doesn't look like a memory leak. Observe for a few days/week and check the memory usage again. Post a screenshot if it keeps growing.

**Reply by Pavel-Husakouski** [reply]
It is most likely a memory leak.  Regardless of the version, netdata tries to occupy all available memory for the container. Then the container restarts. I started from 100mb limit, now I see the very same thing for 400mb. 
So, I decided to stay away from netdata, as the most of the time I see reports about their update and memory. I definitely do not need a flood of false positive reports on the production environment.
